### PR TITLE
Misc documentation

### PR DESCRIPTION
**Pull request by huderlem**

### CODE REVIEW

File: include/constants/field_effects.h [hunk 1]
```diff
@@ -67,6 +67,6 @@
 #define FLDEFF_USE_TELEPORT              63
 #define FLDEFF_64                        64
 #define FLDEFF_65                        65
-#define FLDEFF_66                        66
+#define FLDEFF_MOVE_DEOXYS_ROCK                        66
```
Comment (Diegoisawesome) [inline include/constants/field_effects.h]: Fix value indentation

File: include/constants/flags.h [hunk 2]
```diff
@@ -1356,7 +1356,7 @@
 #define FLAG_SYS_FRONTIER_PASS   (SYSTEM_FLAGS + 0x72)
 
 #define FLAG_0x8D3               (SYSTEM_FLAGS + 0x73)
-#define FLAG_0x8D4               (SYSTEM_FLAGS + 0x74)
+#define FLAG_DEOXYS_ROCK_COMPLETE               (SYSTEM_FLAGS + 0x74)
```
Comment (Diegoisawesome) [inline include/constants/flags.h]: Fix indentation

File: include/constants/vars.h [hunk 3]
```diff
@@ -68,8 +68,8 @@
 #define VAR_FRONTIER_GAMBLER_AMOUNT_BET     0x4032
 #define VAR_FRONTIER_GAMBLER_PLACED_BET_F   0x4033
 
-#define VAR_0x4034                 0x4034
-#define VAR_0x4035                 0x4035
+#define VAR_DEOXYS_ROCK_STEP_COUNT                 0x4034
```
Comment (Diegoisawesome) [inline include/constants/vars.h]: Fix indentation

File: src/field_player_avatar.c [hunk 4]
```diff
@@ -791,7 +791,7 @@ bool8 IsPlayerCollidingWithFarawayIslandMew(u8 direction)
     playerY = object->currentCoords.y;
 
     MoveCoords(direction, &playerX, &playerY);
-    mewObjectId = GetEventObjectIdByLocalIdAndMap(1, 0x39, 0x1A);
+    mewObjectId = GetEventObjectIdByLocalIdAndMap(1, MAP_NUM(FARAWAY_ISLAND_INTERIOR), MAP_GROUP(FARAWAY_ISLAND_INTERIOR));
 
     if (mewObjectId == 16)
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: EVENT_OBJECTS_COUNT

**Comment by DizzyEggg** [comment]
lmao, you included the porymap file again

**Pull request merged**